server/index.js

```diff
@@ -52,6 +51,86 @@ app.get('/api/products/:productId', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+
+  if (!req.session.cartId) {
+    res.status(200).json([]);
+  } else {
+    const sql = `select "c"."cartItemId",
+    "c"."price",
+    "p"."productId",
+    "p"."image",
+    "p"."name",
+    "p"."shortDescription"
+    from "cartItems" as "c"
+    join "products" as "p" using("productId")
+    where "c"."cartId" = $1`;
+    const cartId = req.session.cartId;
+    db.query(sql, [cartId])
+      .then(shoppingCart => {
+        return Object.assign(shoppingCart, cartId);
+      });
```

Comment:
Be sure to respond to the client here. They want an array of cart items only, so be sure to take them out of the result object.

Reply:
Also, don't forget to `.catch()` at the end of the promise chain. If there is an error, we want to pass it to `next()` so that the error prints to our terminal (for debugging) and the client at least receives a `500` response.